### ISSUE TITLE
As a user, I want to browse all card images with known events associated with a single member so that I can get an overview of their card activity.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/mep-django#247]
## notes for testing
Use member browse to find members, click through to their detail page, and then click the cards page.
- You should see thumbnail images for all cards associated with that person (association is based on footnote bibliography record for the lending library card associated with the member account)
- Card images should be labeled by year or year range when we can determine them from the database; years are based on events associated with those cards by footnote in the database; if no years are found, it's labeled as Unknown
-  Dates with unknown years should be ignored when generating year labels (I think Gertrude Stein is currently an example fo this)
- Images are ordered based on the canvas order within the IIIF manifest (some of these probably still need to be edited in Figgy)
- Card tab should be clickable from the membership activities page
- Check at least one member who does not have cards: the card page should still be clickable, but no images will display and there should be text indicating the member does not have any cards available

### design testing
Should display three thumbnails to a row on desktop and tablet; single thumbnail per row on mobile. Try to check a few accounts with cards that aren't multiples of three to check that the last row aligns the way you'd expect.  
Member card gallery in Zeplin: [desktop](https://zpl.io/25oW0mW) | [tablet](https://zpl.io/bWmP1mM) | [mobile](https://zpl.io/VDo9klO)

**Comment by clmahoney** [comment]
Desktop: 
- YES, if a member has a card I am able to see thumbnails by clicking through the member detail page. I checked these members with cards: Pauline Alderman, Walter Benjamin, Antoinette Bernheim. 
QUESTION: Bernheim's cards were out of order, so I went to Figgy and corrected them. But it did not update on the test site. 
- YES, card images are labeled by year or year range, and YES if no years are found, it's labeled as Unknown. 
- YES, if the card has unknown years but dates, like Gertrude Stein's, then the card is labeled as "Unknown"
- YES, I see that the images are based  on the canvas order within the IIIF manifest. And yes, I did have to edit some. But like I said above: it didn't update on the screen. Is that because I have to update the order of the manifests elsewhere? 
- YES, the card tab is clickable from the membership activities page on each member
- YES, if the member does NOT have a card, I am still able to click on the card page and see the text "No cards available." Tried with Ably and Adler.

**Comment by rlskoeser** [comment]
@clmahoney we have to pull in figgy changes, they don't automatically propagate. Do you want me to try updating this one on the test site to check?

**Comment by clmahoney** [comment]
@rlskoeser yes!

**Comment by rlskoeser** [comment]
@clmahoney is the functionality testing resolved based on our test of updating the figgy records you changed? And could you review the styles, also?

**Comment by clmahoney** [comment]
@rlskoeser Yes this is ready to close, but I thought @jkotin would be the one to review and close.

**Comment by jkotin** [comment]
This works well, and is ready to close if @clmahoney has tested on tablet and mobile.

Some notes:

1. a rectangular square surrounds the card and date when cursor touches.  See attached screen grab.

2. We still need to figure out how to handle the blank cards. At the very least, they should not have "unknown" for dates. My current thinking is that we should not show blank cards on the site at all. If a user wants to know recto-verso, the user can go to the finding aid.  Most users will be discouraged/confused by the blanks.

<img width="607" alt="Screen Shot 2019-11-30 at 9 48 56 PM" src="https://user-images.githubusercontent.com/12616674/69908619-7a4e8600-13bb-11ea-94fb-9187bbfa2460.png">

**Comment by clmahoney** [comment]
Tablet: 
- YES, if a member has a card I am able to see thumbnails by clicking through the member detail page. I tested with Bedouret and Pound. 
- YES, card images are labeled by year or year range, and YES if no years are found, it's labeled as Unknown. I have found some mistakes where the year on the site is displaying differently than the year on the card, but I'm making a list of these to share once I've gone through all the cards. 
- YES, if the card has unknown years but does have days and months, like Pound's, the card is listed as "unknown"
- YES, the card tab is clickable from the membership activities page on each member
- YES, if the member does NOT have a card, I am still able to click on the card page and see the text "No cards available."

**Comment by clmahoney** [comment]
Mobile: tested with Arvanon e
- YES, if a member has a card I am able to see thumbnails by clicking through the member detail page. 
- YES, the card images are labeled by year or year range, and YES if no years are found, it's labeled as Unknown.
- YES, if the card has unknown years but does have days and months, the card is listed as "unknown"
- YES, the card tab is clickable from the membership activities page on each member
- YES, if the member does NOT have a card, I am still able to click on the card page and see the text "No cards available."

**Comment by clmahoney** [comment]
I'm ready to close once Josh's two questions have been addressed.

**Comment by thatbudakguy** [comment]
@jkotin re: your comment about the square (a focus indicator) - this one looks maybe a little peculiar, but in general this phenomenon is desirable because it indicates to users where the current focus is. this is particularly helpful for keyboard-only users who use <kbd>tab</kbd> or other keys to navigate around the site and can't always tell where they are "looking" unless there is a visible focus indicator.

**Comment by rlskoeser** [comment]
@jkotin same question as on the other card story - what's your sense of the priority of excluding unknowns, and are you fine with possibly excluding non-blank cards with no footnotes?

**Comment by rlskoeser** [comment]
@jkotin @clmahoney Updated to skip over images without any associated events unless you are logged in. Please test both logged in and not logged in to compare. 

Note that login restrictions to member pages have been turned off in preparation for members MVP and to make it possible to test this feature.

**Comment by jkotin** [comment]
@rlskoeser apologies: how do I explore the cards when I am not logged in?

**Comment by thatbudakguy** [comment]
@jkotin to log out, you can navigate to [the admin site](https://test-shakespeareandco.cdh.princeton.edu/admin/), click your name in the upper right-hand corner, and click "log out". then you can visit the site again to view it while logged out.

**Comment by jkotin** [comment]
@rlskoeser Excellent. This looks good. I am looking at desktop, not logged in. I definitely think we made the right decision suppressing the blank cards. That said, we will have to un-suppress some cards. E.g. @clmahoney Alvear s7—there are no events on the card, but there is an address and the card is actually a card (as opposed to a scrap of paper or an envelop). Such cards should be part of the site. This isn't a blocker for this release, but I would like to have a list of cards that will be need to be "unsuppressed" for January 1 to include in the next release. Is this reasonable for everyone?  I found the problem by comparing the site to the finding aid on two screens, side by side.
